src/modm/platform/gpio/lpc/unused.hpp.in

```diff
@@ -0,0 +1,121 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_LPC_GPIO_PIN_UNUSED_HPP
+#define MODM_LPC_GPIO_PIN_UNUSED_HPP
+
+#include "base.hpp"
+#include <modm/architecture/interface/gpio.hpp>
+
+namespace modm
+{
+
+namespace platform
+{
+
+/**
+ * Dummy implementation of an I/O pin.
+ *
+ * This class can be used when a pin is not required. All functions
+ * are dummy functions which do nothing. `read()` will always
+ * return `false`.
+ *
+ * For example when creating a software SPI with the modm::SoftwareSimpleSpi
+ * class and the return channel (MISO - Master In Slave Out) is not needed,
+ * a good way is to use this class as a parameter when defining the
+ * SPI class.
+ *
+ * Example:
+ * @code
+ * #include <modm/architecture/platform.hpp>
+ *
+ * namespace pin
+ * {
+ *     typedef GpioOutputD7 Clk;
+ *     typedef GpioOutputD5 Mosi;
+ * }
+ *
+ * modm::SoftwareSpiMaster< pin::Clk, pin::Mosi, GpioUnused > Spi;
+ *
+ * ...
+ * Spi::write(0xaa);
+ * @endcode
+ *
+ * @author	Fabian Greif
+ * @author	Niklas Hauser
+ * @ingroup	modm_platform_gpio
+ */
+class GpioUnused : public Gpio, public ::modm::GpioIO
+{
+public:
+	using Output = GpioUnused;
+	using Input = GpioUnused;
+	using IO = GpioUnused;
+	using Type = GpioUnused;
+	static constexpr bool isInverted = false;
+	static constexpr Port port = Port(-1);
+	static constexpr uint8_t pin = uint8_t(-1);
+	static constexpr uint16_t mask = 0;
+
+protected:
+	/// @cond
+	static void setAlternateFunction(uint8_t) {}
+	static void setAnalogInput() {}
+	/// @endcond
+
+public:
+	// GpioOutput
+	// start documentation inherited
+	static void setOutput() {}
+	static void setOutput(bool) {}
+	static void set() {}
+	static void set(bool) {}
+	static void reset() {}
+	static void toggle() {}
+	static bool isSet() { return false; }
+	// stop documentation inherited
+	static void configure(OutputType) {}
+	static void setOutput(OutputType) {}
+
+	// GpioInput
+	// start documentation inherited
+	static void setInput() {}
+	static bool read() { return false; }
+	// end documentation inherited
+	static void configure(InputType) {}
+	static void setInput(InputType) {}
+	// External Interrupts
+	static void enableExternalInterrupt() {}
+	static void disableExternalInterrupt() {}
+	static void enableExternalInterruptVector(const uint32_t) {}
+	static void disableExternalInterruptVector() {}
+	static void setInputTrigger(const InputTrigger) {}
+	static bool getExternalInterruptFlag() { return false; }
+	/// Reset the interrupt flag in the interrupt routine.
+	static void acknowledgeExternalInterruptFlag() {}
+
+	// GpioIO
+	// start documentation inherited
+	static Direction getDirection() { return Direction::Special; }
+	// end documentation inherited
+	static void lock() {}
```

Comment:
This needs to have the same API as the platform specific GPIO classes. It should be drop-in replacable.